src/coreclr/jit/compiler.cpp

```diff
@@ -5265,16 +5250,8 @@ static bool HasOldChildLoop(Compiler* comp, FlowGraphNaturalLoop* loop)
 //
 bool Compiler::shouldAlignLoop(FlowGraphNaturalLoop* loop, BasicBlock* top)
 {
-    // TODO-Quirk: Remove. When removing we will likely need to add some
-    // form of "lexicality" heuristic here: only align loops whose blocks
-    // are fairly tightly packed together physically.
```

Comment:
I decided not to look at this here; in particular because we have a lot of work in flight that is severely changing the flow graph, so it's hard to get a good feeling for how this heuristic should eventually look.